### PR TITLE
Enhancement: Disable LRU `flushPendingWrite` warning if disabled

### PR DESCRIPTION
**Pull request by ahangsu**
## Summary

Per Pavel's request, disable `flushPendingWrite` warning if LRU cache is disabled.

### CODE REVIEW

File: ledger/lruaccts.go [hunk 1]
```diff
@@ -80,6 +80,9 @@ func (m *lruAccounts) readNotFound(addr basics.Address) bool {
 // thread locking semantics : write lock
 func (m *lruAccounts) flushPendingWrites() {
 	pendingEntriesCount := len(m.pendingAccounts)
+	if pendingEntriesCount == 0 {
+		return
+	}
```
Comment (ahangsu) [inline ledger/lruaccts.go:85]: WLOG take this as example: if `pendingEntriesCount` is length 0, nothing to flush in, fast exit; when the LRU cache is disabled, `m.pendingAccounts` can only be length 0, so we exit early before warning.

File: agreement/types.go [hunk 2]
```diff
@@ -68,9 +68,9 @@ const (
 )
```
Comment (algorandskiy) [inline agreement/types.go:68]: this file should not be changed in this PR

**Review by algorandskiy — DISMISSED** [review]

File: agreement/types.go (hunk 2 above)
Comment (ahangsu) [reply agreement/types.go:68]: ok, reverting this and separate out for another PR then

**Review by jannotti — COMMENTED** [review]
I would rather see the calls that initiialise the caches in disabled mode, which currently look like this:
```
		au.baseAccounts.init(au.log, 0, 0)
		au.baseResources.init(au.log, 0, 0)
		au.baseKVs.init(au.log, 0, 0)
```
changed to
```
		au.baseAccounts.init(au.log, 0, 1)
		au.baseResources.init(au.log, 0, 1)
		au.baseKVs.init(au.log, 0, 1)
```
which implicitly turns off the warning, since the threshold is higher than the size.
That avoids the extra test all the time.

**Comment by ahangsu** [comment]
yea, that's fair, I will flip the change asap.

**Review by algorandskiy — APPROVED** [review]

**Review by jannotti — APPROVED** [review]

**Pull request merged**